src/Markdig/Extensions/Emoji/EmojiMapping.cs

```diff
@@ -0,0 +1,1784 @@
+// Copyright (c) Alexandre Mutel. All rights reserved.
+// This file is licensed under the BSD-Clause 2 license. 
+// See the license.txt file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Markdig.Helpers;
+
+namespace Markdig.Extensions.Emoji
+{
+    /// <summary>
+    /// An emoji and smiley mapping, to be used by <see cref="EmojiParser"/>.
+    /// </summary>
+    public class EmojiMapping
+    {
+        /// <summary>
+        /// The default emojis and smileys mapping.
+        /// </summary>
+        public static readonly EmojiMapping DefaultEmojiMapping = new EmojiMapping();
+
+        internal CompactPrefixTree<string> EmojiPrefixTree { get; }
+        internal CompactPrefixTree<string> EmojiSmileyPrefixTree { get; }
+        internal char[] EmojiOpeningCharacters { get; }
+        internal char[] EmojiSmileyOpeningCharacters { get; }

```

Comment:
Nit: readonly

Reply:
You mean a `readonly` field instead of a read-only property? I can't just add `readonly` here.

Why do you want that? There is a warning against it https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2105

Reply:
Mb, missed that you made it a property